elasticsearch/templates/statefulset.yaml

```diff
@@ -295,15 +295,12 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.name
-          {{- if eq .Values.roles.master "true" }}
```

Comment:
This is coming from the Helm template right? Not elasticsearch.yml? So effectively we've changed how you specify roles to only support the new `node.roles` setting? If so, how do we handle compatibility with versions of Elasticsearch that don't have the `node.roles` setting?

Reply:
Yes this is coming from the Helm template to define some environment variables into Elasticsearch container.

>If so, how do we handle compatibility with versions of Elasticsearch that don't have the node.roles setting?

helm-charts repo is using release branches following the same model as [elastic/elasticsearch](https://github.com/elastic/elasticsearch/). We expect to merge this PR only on `master` branch so it will only be released with helm-charts 8.x which will use Elasticsearch 8.x Docker images.

Currently master branch is using [8.0.0-SNAPSHOT image](https://github.com/elastic/helm-charts/blob/master/elasticsearch/values.yaml#L62) which is why CI tests is failing, while all [released versions](https://github.com/elastic/helm-charts/releases] are using aligned Elasticsearch versions.

Note that anybody can override the `imageTag` value, to use the chart from master branch with an old Elasticsearch version for example. However this is not something we support and we don't try to handle backward compatibility in this scenario.